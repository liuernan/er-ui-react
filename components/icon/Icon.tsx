import React from 'react';
import './importIcons';
import '../../styles/index.scss';
import classNames from '../../helpers/classnames';

interface IconProps extends React.SVGAttributes<SVGElement> {
  name: string
}

const Icon: React.FunctionComponent<IconProps> = ({name, className, ...restProps}) => {
  return (
    <svg className={classNames('er-ui-icon', className)}
         {...restProps}
    >
      <use xlinkHref={`#${name}`}/>
    </svg>
  );
};

export default Icon;