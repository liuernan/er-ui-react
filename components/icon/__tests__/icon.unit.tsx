import React from 'react';
import * as renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import Icon from '../Icon';

describe('Icon', () => {
  it('rendering successfully', () => {
    const json = renderer.create(<Icon name="heart"/>).toJSON();
    expect(json).toMatchSnapshot();
  });

  it('has custom class names', () => {
    const json = renderer.create(<Icon name="heart" className="custom custom2"/>).toJSON();
    expect(json).toMatchSnapshot();
  });

  it('can handling events', () => {
    const fn = jest.fn();
    const component = mount(<Icon name="alipay" onClick={fn}/>);
    component.find('svg').simulate('click');
    expect(fn).toBeCalled();
  });
});