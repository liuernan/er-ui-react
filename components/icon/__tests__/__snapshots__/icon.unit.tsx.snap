// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon has custom class names 1`] = `
<svg
  className="er-ui-icon custom custom2"
>
  <use
    xlinkHref="#heart"
  />
</svg>
`;

exports[`Icon rendering successfully 1`] = `
<svg
  className="er-ui-icon"
>
  <use
    xlinkHref="#heart"
  />
</svg>
`;

exports[`icon has custom class names 1`] = `
<svg
  className="er-ui-icon custom custom2"
>
  <use
    xlinkHref="#heart"
  />
</svg>
`;

exports[`icon render successfully 1`] = `
<svg
  className="er-ui-icon"
>
  <use
    xlinkHref="#heart"
  />
</svg>
`;
